### ISSUE TITLE
late fusion of these two separate heads, should add the softmax output

### ISSUE THREAD

**Issue by vivekh2000** [FrancescoSaverioZuppichini/DeiT#1]
![image](https://github.com/FrancescoSaverioZuppichini/DeiT/assets/90518409/621ccc2d-de7a-4572-b25f-837461c8d481)

Thanks for the nice article and neat code.
In your article you mentioned that DeiT-S is 100x faster than ViT-B. However, looking at the graph, it seems to me that it is 10x faster.
Also, in the ClassificationHead class, you have taken the average of class token logits and distillation token logits at the inference time. However, if you look at the original paper, the authors say that
" **_Yet our referent method is the late fusion of these two separate heads, for which we add the softmax output by the two classifiers to make the prediction_**. " (p. 8, third paragraph)
It is clear from the above statement that authors have taken an average after the softmax steps, not after the logits. Please correct me if I am wrong. :)
